opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexEnumerator.java

```diff
@@ -135,6 +125,7 @@ public boolean moveNext() {
 
   @Override
   public void reset() {
+    bgScanner.reset(request);
     OpenSearchResponse response = client.search(request);
```

Comment:
Why use client.search instead of bgScanner.fetchNextBatch in this case?

Reply:
Since it's the same client in both instances, it doesn't make a difference (fetchNextBatch would force a sync fetch). I can change it for consistency